pkg/processor/runtime/types.go

```diff
@@ -17,42 +17,90 @@ limitations under the License.
 package runtime
 
 import (
-	"github.com/nuclio/nuclio/pkg/util/common"
+	"github.com/nuclio/nuclio/pkg/functioncr"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/viper"
+	"os"
+	"strings"
 )
 
-// TODO: this should be a whole thing. for now this should do
-type DataBinding struct {
-	Class string
-	URL   string
-}
-
 type Configuration struct {
 	Name         string
 	Version      string
 	Description  string
-	DataBindings []DataBinding
+	DataBindings map[string]*functioncr.DataBinding
 }
 
 func NewConfiguration(configuration *viper.Viper) (*Configuration, error) {
 
 	newConfiguration := &Configuration{
-		Name:        configuration.GetString("name"),
-		Description: configuration.GetString("description"),
-		Version:     configuration.GetString("version"),
+		Name:         configuration.GetString("name"),
+		Description:  configuration.GetString("description"),
+		Version:      configuration.GetString("version"),
+		DataBindings: map[string]*functioncr.DataBinding{},
 	}
 
-	// read data bindings
-	dataBindings := common.GetObjectSlice(configuration, "data_bindings")
-	for _, dataBinding := range dataBindings {
-		newDataBinding := DataBinding{
-			Class: dataBinding["class"].(string),
-			URL:   dataBinding["url"].(string),
-		}
+	// get databindings by environment variables
+	err := newConfiguration.getDataBindingsFromEnv(os.Environ(), newConfiguration.DataBindings)
 
-		newConfiguration.DataBindings = append(newConfiguration.DataBindings, newDataBinding)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to read data bindings from environment")
 	}
 
 	return newConfiguration, nil
 }
+
+func (c *Configuration) getDataBindingsFromEnv(envs []string, dataBindings map[string]*functioncr.DataBinding) error {
+	dataBindingPrefix := "NUCLIO_DATA_BINDING_"
+
+	// iterate over env
+	for _, env := range envs {
+		envKeyValue := strings.Split(env, "=")
+		envKey := envKeyValue[0]
+		envValue := envKeyValue[1]
+
+		// check if it starts with data binding prefix. if it doesn't do nothing
+		if !strings.HasPrefix(envKey, dataBindingPrefix) {
+			continue
+		}
+
+		// strip the prefix
+		envKey = envKey[len(dataBindingPrefix):]
+
+		// look for the known postfixes
+		for _, postfix := range []string{
```

Comment:
Use a regular expression here?

```go
	re := regexp.MustCompile("^NUCLIO_DATA_BINDING_(.+)=(.+)$")
	envs := []string{
		"NUCLIO_DATA_BINDING_A=AVAL",
		"NOT_NUCLIO_DATA_BINDING_A=AVAL",
		"NUCLIO_DATA_BINDING_=Nope",
		"NUCLIO_DATA_BINDING_B=BVAL",
	}
	for _, env := range envs {
		matches := re.FindStringSubmatch(env)
		if len(matches) == 0 {
			continue
		}
		fmt.Printf("%q -> %q\n", matches[1], matches[2])
	}
```

Reply:
Would require two regexes - one to capture the <name>_<CLASS/URL> and another to extract the <name> and CLASS/PATH. I always prefer regexes over manual parsing but here it's overcomplicating